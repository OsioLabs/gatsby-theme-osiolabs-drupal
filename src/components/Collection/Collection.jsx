import React from 'react';
import PropTypes from 'prop-types';
import { Container, Segment } from 'semantic-ui-react';
import TutorialList from '../Tutorial/TutorialList';

const Collection = ({ summary, tutorials }) => (
  <Container>
    <div dangerouslySetInnerHTML={{ __html: summary }} />

    {tutorials && tutorials.length > 0 && (
      <Segment>
        <h2>Tutorials in this collection:</h2>
        <TutorialList tutorials={tutorials} />
      </Segment>
    )}
  </Container>
);

Collection.propTypes = {
  // eslint-disable-next-line
  id: PropTypes.string.isRequired,
  // eslint-disable-next-line
  title: PropTypes.string.isRequired,
  // eslint-disable-next-line
  summary: PropTypes.string.isRequired,
  tutorials: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
      path: PropTypes.string.isRequired,
      tutorial_access: PropTypes.string.isRequired,
    })
  ).isRequired,
};

export default Collection;
