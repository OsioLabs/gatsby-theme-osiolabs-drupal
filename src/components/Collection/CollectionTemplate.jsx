import React from 'react';
import Helmet from 'react-helmet';
import Collection from './Collection';

/**
 * This is a super simple example of a CollectionTemplate component.
 *
 * In most cases you'll probably want to override this with a site specific
 * component.
 */
const CollectionTemplate = ({ data }) => {
  const {
    // eslint-disable-next-line camelcase
    drupal_id,
    relationships,
    // eslint-disable-next-line camelcase
    collection_summary,
    title,
  } = data.nodeCollection;

  let tutorials = [];
  if (relationships.tutorials && relationships.tutorials.length > 0) {
    tutorials = relationships.tutorials.map(item => ({
      title: item.title,
      id: item.drupal_id,
      path: item.path.alias,
      tutorial_access: item.tutorial_access,
    }));
  }

  return (
    <div className="collection">
      <Helmet title={`${title} | Gatsby Guides`} />
      <Collection
        id={drupal_id}
        title={title}
        summary={collection_summary.processed}
        tutorials={tutorials}
      />
    </div>
  );
};

export default CollectionTemplate;
