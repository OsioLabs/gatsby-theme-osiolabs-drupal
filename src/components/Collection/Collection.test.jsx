import React from 'react';
import { render } from '@testing-library/react';

import Collection from './Collection';

const tutorials = [
  {
    id: 'one',
    title: 'Tutorial One',
    path: '/tutorial/one',
    tutorial_access: 'public',
  },
  {
    id: 'two',
    title: 'Tutorial Two',
    path: '/tutorial/two',
    tutorial_access: 'membership_required',
  },
  {
    id: 'three',
    title: 'Tutorial Three',
    path: '/tutorial/three',
    tutorial_access: 'asdf',
  },
];

describe('Component: <Collection />', () => {
  it('renders correctly', () => {
    // Basic snapshot test.
    const { asFragment } = render(
      <Collection
        title="Tacos Rock"
        path="/tutorial/tacos"
        id="collection-id"
        summary="<p>Copy goes here ...</p>"
        tutorials={tutorials}
      />
    );
    const firstRender = asFragment();
    expect(firstRender).toMatchSnapshot();
  });
});
