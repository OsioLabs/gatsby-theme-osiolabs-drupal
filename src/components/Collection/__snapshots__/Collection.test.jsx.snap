// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <Collection /> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="ui container"
  >
    <div>
      <p>
        Copy goes here ...
      </p>
    </div>
    <div
      class="ui segment"
    >
      <h2>
        Tutorials in this collection:
      </h2>
      <div
        class="ui items"
      >
        <div
          class="item tutorial--item"
        >
          <div
            class="content"
          >
            <div
              class="ui circular label"
            >
              1
            </div>
          </div>
          <div
            class="content"
          >
            <div
              class="header"
            >
              <a
                href="/tutorial/one"
              >
                Tutorial One
              </a>
            </div>
            <div
              class="meta"
            >
              <i
                aria-label="Tutorial contains text"
                class="file text icon"
              />
              <i
                aria-label="Tutorial has one or more videos"
                class="video play icon"
              />
               3 minutes
            </div>
          </div>
          <div
            class="content"
          >
            <i
              aria-label="Membership required. Preview available"
              class="lock large icon"
            />
          </div>
        </div>
        <div
          class="item tutorial--item"
        >
          <div
            class="content"
          >
            <div
              class="ui circular label"
            >
              2
            </div>
          </div>
          <div
            class="content"
          >
            <div
              class="header"
            >
              <a
                href="/tutorial/two"
              >
                Tutorial Two
              </a>
            </div>
            <div
              class="meta"
            >
              <i
                aria-label="Tutorial contains text"
                class="file text icon"
              />
              <i
                aria-label="Tutorial has one or more videos"
                class="video play icon"
              />
               3 minutes
            </div>
          </div>
          <div
            class="content"
          >
            <i
              aria-label="Membership required. Preview available"
              class="lock large icon"
            />
          </div>
        </div>
        <div
          class="item tutorial--item"
        >
          <div
            class="content"
          >
            <div
              class="ui circular label"
            >
              3
            </div>
          </div>
          <div
            class="content"
          >
            <div
              class="header"
            >
              <a
                href="/tutorial/three"
              >
                Tutorial Three
              </a>
            </div>
            <div
              class="meta"
            >
              <i
                aria-label="Tutorial contains text"
                class="file text icon"
              />
              <i
                aria-label="Tutorial has one or more videos"
                class="video play icon"
              />
               3 minutes
            </div>
          </div>
          <div
            class="content"
          >
            <i
              aria-label="Membership required. Preview available"
              class="lock large icon"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
